rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -318,21 +320,27 @@ public void saveLockWhitelist() {
             return;
         }
 
-        safeSaveToRepository(LOCK_WHITELIST_KEY, lockWhitelist, BridgeSerializationUtils::serializeLockWhitelist);
+        safeSaveToRepository(LOCK_ONE_OFF_WHITELIST_KEY, lockWhitelist, BridgeSerializationUtils::serializeOneOffLockWhitelist);
+
+        safeSaveToRepository(LOCK_UNLIMITED_WHITELIST_KEY, lockWhitelist, BridgeSerializationUtils::serializeUnlimitedLockWhitelist);
```

Comment:
This would fork the storage in a way we don't want to (i.e., this would be a HARD FORK), given that we always write the unlimited part regardless.

Reply:
I will add a configuration class to the storageProvider which will include the state of the unlimited whitelist. If it's enabled then I'll call the safesave (same with the gets) otherwise I'll ignore those methods.